eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java

```diff
@@ -0,0 +1,245 @@
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.dbsnp.io;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.commons.accession.core.models.EventType;
+import uk.ac.ebi.ampt2d.commons.accession.hashing.SHA1HashingFunction;
+
+import uk.ac.ebi.eva.accession.core.ClusteredVariant;
+import uk.ac.ebi.eva.accession.core.IClusteredVariant;
+import uk.ac.ebi.eva.accession.core.ISubmittedVariant;
+import uk.ac.ebi.eva.accession.core.SubmittedVariant;
+import uk.ac.ebi.eva.accession.core.configuration.MongoConfiguration;
+import uk.ac.ebi.eva.accession.core.configuration.SubmittedVariantAccessioningConfiguration;
+import uk.ac.ebi.eva.accession.core.persistence.DbsnpSubmittedVariantEntity;
+import uk.ac.ebi.eva.accession.core.persistence.DbsnpSubmittedVariantInactiveEntity;
+import uk.ac.ebi.eva.accession.core.persistence.DbsnpSubmittedVariantOperationEntity;
+import uk.ac.ebi.eva.accession.core.summary.DbsnpClusteredVariantSummaryFunction;
+import uk.ac.ebi.eva.accession.core.summary.DbsnpSubmittedVariantSummaryFunction;
+import uk.ac.ebi.eva.accession.dbsnp.persistence.DbsnpClusteredVariantEntity;
+import uk.ac.ebi.eva.accession.dbsnp.persistence.DbsnpVariantsWrapper;
+import uk.ac.ebi.eva.accession.dbsnp.test.MongoTestConfiguration;
+import uk.ac.ebi.eva.commons.core.models.VariantType;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@TestPropertySource("classpath:test-variants-writer.properties")
+@ContextConfiguration(classes = {MongoConfiguration.class, MongoTestConfiguration.class,
+        SubmittedVariantAccessioningConfiguration.class})
+public class DbsnpVariantsWriterTest {
+
+    private static final int TAXONOMY_1 = 3880;
+
+    private static final int TAXONOMY_2 = 3882;
+
+    private static final long EXPECTED_ACCESSION = 10000000000L;
+
+    private static final int START_1 = 100;
+
+    private static final Long CLUSTERED_VARIANT = 12L;
+
+    private static final VariantType VARIANT_TYPE = VariantType.SNV;
+
+    private static final Long SUBMITTED_VARIANT = 15L;
+
+    private static final Boolean SUPPORTED_BY_EVIDENCE = true;
+
+    private static final Boolean MATCHES_ASSEMBLY = false;
+
+    private static final Boolean ALLELES_MATCH = true;
+
+    private static final Boolean VALIDATED = false;
```

Comment:
If using the defaults would make no difference, it could be nicer to just use those from `ISubmittedVariant`.